### ISSUE TITLE
Build a Small-CNN for Stop Sign Detection

### ISSUE THREAD

**Issue by obaidmm** [WE-Autopilot/WEAP_CV#28]
In this task, we will design and implement a Convolutional Neural Network (CNN) for real-time stop sign detection.

The goal is to build a lightweight model that can accurately detect stop signs in images while ensuring it runs efficiently on low-power. The model should be able to process images. To achieve this, we will construct a small CNN with convolutional layers for feature extraction, batch normalization for stable training, pooling layers to reduce spatial dimensions, and fully connected layers for classification **(You should do more research on this ).** The final layer will use a Softmax activation function to output the probability of a stop sign being present.

The final deliverable for this task will be a Python script (cnn_detectionmodel.py) that defines the model, loading and preprocesses the dataset, and saves the trained model in an appropriate format (basically .pt).
